src/main/java/com/overops/plugins/service/impl/ReportBuilder.java

```diff
@@ -366,83 +366,83 @@ public static QualityReport execute(ApiClient apiClient, RegressionInput input,
 			uniqueEventCount = 0;
 			maxUniqueErrorsExceeded = false;
 		}
-		
+
 		//new error gate
 		boolean newErrors = false;
 		if (qualityGateReport.getNewErrors() != null && qualityGateReport.getNewErrors().size() > 0) {
 			newErrors = true;
 			replaceSourceId(qualityGateReport.getNewErrors());
 		}
-		
+
 		//resurfaced error gate
 		boolean resurfaced = false;
 		if (qualityGateReport.getResurfacedErrors() != null && qualityGateReport.getResurfacedErrors().size() > 0) {
 			resurfaced = true;
 			replaceSourceId(qualityGateReport.getResurfacedErrors());
 		}
-		
+
 		//critical error gate
 		boolean critical = false;
 		if (qualityGateReport.getCriticalErrors() != null  && qualityGateReport.getCriticalErrors().size() > 0) {
 			critical = true;
 			replaceSourceId(qualityGateReport.getCriticalErrors());
 		}
-		
+
 		//top errors
 		if (qualityGateReport.getTopErrors() != null  && qualityGateReport.getTopErrors().size() > 0) {
 			replaceSourceId(qualityGateReport.getTopErrors());
 		}
-		
+
 		boolean checkCritical = false;
 		if (input.criticalExceptionTypes != null && input.criticalExceptionTypes.size() > 0) {
 			checkCritical = true;
 		}
-		
+
 		boolean unstable = (failedAPI)
 				|| (hasRegressions)
 				|| (maxVolumeExceeded)
 				|| (maxUniqueErrorsExceeded)
 				|| (newErrors)
 				|| (resurfaced)
 				|| (critical);
-		
+
 		return new QualityReport(input, rateRegression, regressions,
-				qualityGateReport.getCriticalErrors(), qualityGateReport.getTopErrors(), qualityGateReport.getNewErrors(), 
+				qualityGateReport.getCriticalErrors(), qualityGateReport.getTopErrors(), qualityGateReport.getNewErrors(),
 				qualityGateReport.getResurfacedErrors(), qualityGateReport.getTotalErrorCount(),
 				qualityGateReport.getUniqueErrorCount(), unstable, newEvents, resurfacedEvents, checkCritical, checkMaxEventGate,
 				checkUniqueEventGate, runRegressions, maxEventVolume, maxUniqueErrors, markedUnstable);
 	}
-	
+
 	//for each event, replace the source ID in the ARC link with the number 4 (which means Jenkins)
 	private static void replaceSourceId (List<OOReportEvent> events) {
```

Comment:
see https://github.com/jenkinsci/overops-query-plugin/commit/d8447e7598c169b115921a3e271b96d0d9b12aa4

please combine `replaceSourceId`, `replaceSourceId2`, and `replaceSourceIdInArcLink` with a single method.

Reply:
done